hal-sim/hal_impl/functions.py

```diff
@@ -1475,23 +1482,51 @@ def getUserCurrentFaults3V3(status):
 # Relay.h
 #############################################################################
 
+def _handle_to_channel(relayPortHandle):
+    channel = int(relayPortHandle.pin/2)
+    return channel, relayPortHandle.pin % 2 == 0
+
 def initializeRelayPort(portHandle, fwd, status):
     status.value = 0
-    assert False
+    pin = portHandle.pin * 2
+    if not fwd:
+        pin = pin + 1
+        hal_data['relay'][portHandle.pin]['rev'] = False
+    else:
+        hal_data['relay'][portHandle.pin]['fwd'] = False
+
+    hal_data['relay'][portHandle.pin]['initialized'] = True
+
+    return types.RelayHandle(pin)
 
 def freeRelayPort(relayPortHandle):
-    assert False
+    channel, fwd = _handle_to_channel(relayPortHandle)
+    if fwd:
+        hal_data['relay'][channel]['fwd'] = False
+    else:
+        hal_data['relay'][channel]['rev'] = False
 
+    
 def checkRelayChannel(channel):
-    assert False
+    return 0 <= channel and channel < kNumRelayHeaders
```

Comment:
Sometimes I wonder whether we should just use chained comparisons in these check functions…